Console/Command/CakePHPUtilShell.php

```diff
@@ -0,0 +1,330 @@
+		if (!$this->_debug) {
+			set_error_handler(array(new AppShellErrorHandler(), "warning_handler"), E_ALL);
+		}
+	}
+
+/**
+ * This method allows you to easily convert arguments into variables for your shell.
+ * It accepts two input arrays, arguments and defaults, which are described below.
+ * If the defaults array is not set, the default will be set to null for all args that are not passed.
+ *
+ * @author Jan Dorsman <jdorsman@expandonline.nl>
+ *
+ * @param array $args     The name you want the arguments to have.
+ * @param array $defaults The default value you want the arguments to have if no value was passed for it.
+ *
+ * Example 1: $args = $this->processArgs(array('someVar', 'somethingElse', 'userId'));
+ * Would set the vars: $someVar, $somethingElse and $userId with the values of $this->args[0], [1] and [2].
+ * If the specific arg was not passed, it will be set to null.
+ *
+ * Example 2: $args = $this->processArgs(array('letter', 'symbol', 'userId'), array('a', '-', 14));
+ * Would set the $letter, $symbol and $userId vars with the values of $this->args[0], [1] and [2].
+ * If the args were not passed by the user, they would default to: $letter = 'a', $symbol = '-' and $userId = 14;
+ *
+ * Finally, in your shell, run this small foreach loop to set the vars:
+ * foreach ($args as $arg => $val) {
+ *     ${$arg} = $val;
+ * }
+ *
+ * Or alternatively if there are only a few vars:
+ * list($someVar, $somethingElse, $userId) = array_values($args);
+ *
+ * @return array
+ */
+	public function processArgs(array $args, array $defaults = null) {
+		$vars = array();
+		foreach ($args as $id => $arg) {
+			if (array_key_exists($id, $this->args)) {
+				$vars[$arg] = $this->args[$id];
+			} else {
+				if (is_array($defaults) && array_key_exists($id, $defaults)) {
+					$vars[$arg] = $defaults[$id];
+				} else {
+					$vars[$arg] = null;
+				}
+			}
+		}
+
+		return $vars;
+	}
+
+	/**
+	 * Acts as a callback function which will be called if the SubCommand does not exist
+	 */
+	public function main() {
+		if (isset($this->command)) {
+			if (!$this->hasMethod($this->command)) {
+				$this->_exceptions[] = sprintf('The SubCommand \'%s\' you are tyring to execute does not exist!', $this->command);
+			} else {
+				$this->_exceptions[] = sprintf('For some reason this SubCommand \'%s]\' could not be ran', $this->command);
+			}
+		} else {
+			$this->_exceptions[] = 'you have to provide a SubCommand to execute!';
+		}
+
+		#TODO uncomment the line below, which echoes the help. This is used to display the getOptionsParser. Every Shell should be reformatted to the new structure first!
+		#For an example: have a look at the FeedShell
+		$this->out('-- The Help will be enabled when all Shells are restructured to the new layout -- AppShell line ' . (__LINE__ + 1));
+//        $this->out($this->getOptionParser()->help());
+		# ------------------------------------------------------------------------------------------------------------ #
```

Comment:
![MINOR](https://sonarsource.github.io/sonar-github/severity-minor.png 'Severity: MINOR') Use "//" instead of "#" to start this comment [![rule](https://sonarsource.github.io/sonar-github/rule.png)](http://dev.expandonline.nl:9000/sonarqube/coding_rules#rule_key=php%3AS2046)